README.md

```diff
@@ -17,6 +17,74 @@ to be used with [Gluster-Kubernetes](https://github.com/jarrpa/gluster-kubernete
 Our [command flow diagram](docs/diagram/control-diag.md) shows where Kubernetes, service-catalog,
 and the (often external) service broker interact. Service-catalog terms used below are shown in the diagram.
 
+### Nomenclature
+
+Before going further, it's useful to define common terms used throughout the entire system.
+The relationships between the different systems and objects can be seen in the [control flow diagram](docs/diagram/control-diag.md)
+There are a number of naming collisions which can lead to some confusion.
+
+#### External Service Provider
+
+- *External Service Provider*
+
+For our purposes, an External Service Provider is any system that uses a Service Broker to make Services available on demand.  
+The actual location of the External Service Provider is arbitrary.  
+It can be a remote service provider like AWS S3, an on premise cluster, or the same cluster as the Service-Catalog.
+
+The External Service Provider should consist of two components: the service broker and the actual services being served to clients.  In this example, the service component is a CNS Object Storage cluster with a Swift/S3 REST interface.
+
+- *Broker*
+
+The Broker binary is run inside a pod.  The broker implements a server that listens for REST requests and translates them to the underlying Broker methods.  When methods complete, they return OpenServiceBroker responses in json format.
+
+##### Broker Objects
+
+- *ServiceInstance*
+
+  Broker's internal representation of a provisioned service.
+
+- *ServiceInstanceCredential*
+
+  Broker's data structure for tracking coordinates and auth credentials for a single *ServiceInstance*.  
+
+- *Catalog*
+
+  A complete list of services and plans that are provisionable by the Broker.
+
+#### Local Kubernetes Cluster
```

Comment:
I'm confused by the sub-headings here. I was thinking there would be two main headings (per the diagram):
1. External Service Provider (SaaS)
2. K8s Cluster
Then under 1) would be:
   - Service Broker/Deployment
   - Catalog
And under 2) would be:
   - ServiceBroker
   - ServiceClass
   - ServiceInstance
   - ServiceInstanceCredential

Reply:
Several of the service-catalog resources are repeated.

Reply:
The headings are written pretty much as you've laid it out.  Here is how it is organized.
1. External Service Provider
- Broker Objects
  - ServiceInstance
  - ServiceInstanceCredential
  - Catalog
2. Local Kubernetes Cluster (Now "K8s Cluster")
- Service-Catalog API Objects
  - ServiceBroker
  - ServiceClass
  - ServiceInstance
  - ServiceInstanceCredential

I wanted to separate definition and deployment.  Writing them together I think creates confusion and makes it more difficult to follow deployment steps.  This way, the Overview sections functions as an intro to the system, nomenclature, purpose, limitations, and dependencies.